packages/code-editor/src/testing/getTestUtils.spec.tsx

```diff
@@ -0,0 +1,479 @@
+
+      // Render a simple editor for API demonstration
+      render(
+        <CodeEditor
+          defaultValue="console.log('test');"
+          data-lgid="lg-api-demo"
+        />,
+      );
+
+      const utils = getTestUtils('lg-api-demo');
+
+      // Document all available methods
+      expect(typeof utils.getEditor).toBe('function');
+      expect(typeof utils.waitForInitialization).toBe('function');
+      expect(typeof utils.waitForLoadingToComplete).toBe('function');
+      expect(typeof utils.getContent).toBe('function');
+      expect(typeof utils.getLanguage).toBe('function');
+      expect(typeof utils.getIsLoading).toBe('function');
+      expect(typeof utils.getIsReadOnly).toBe('function');
+      expect(typeof utils.getHasLineNumbers).toBe('function');
+      expect(typeof utils.getHasLineWrapping).toBe('function');
+      expect(typeof utils.getHasCodeFolding).toBe('function');
+      expect(typeof utils.getCopyButton).toBe('function');
+      expect(typeof utils.queryPanel).toBe('function');
+      expect(typeof utils.getPanelUtils).toBe('function');
+
+      // Basic synchronous operations work immediately
+      expect(utils.getEditor()).toBeInTheDocument();
+      expect(utils.getLanguage()).toBeNull(); // No language specified
+      expect(typeof utils.getIsLoading()).toBe('boolean');
+    });
+
+    test('Basic synchronous usage example', () => {
+      // Example: Testing basic editor presence and attributes
+      render(
+        <CodeEditor
+          defaultValue="const x = 42;"
+          language={LanguageName.javascript}
+          data-lgid="lg-basic-example"
+        />,
+      );
+
+      const utils = getTestUtils('lg-basic-example');
+
+      // Verify editor is present
+      expect(utils.getEditor()).toBeInTheDocument();
+      expect(utils.getEditor()).toHaveAttribute(
+        'data-lgid',
+        'lg-basic-example',
+      );
+
+      // Test language detection
+      expect(utils.getLanguage()).toBe(LanguageName.javascript);
+
+      // Test loading state (synchronous check)
+      const isLoading = utils.getIsLoading();
+      expect(typeof isLoading).toBe('boolean');
+    });
+
+    test('Loading state testing example', () => {
+      // Example: Testing explicit loading states
+      render(
+        <CodeEditor
+          defaultValue="print('hello')"
+          isLoading={true}
+          data-lgid="lg-loading-example"
+        />,
+      );
+
+      const utils = getTestUtils('lg-loading-example');
+
+      // Should be loading when isLoading prop is true
+      expect(utils.getIsLoading()).toBe(true);
+    });
+
+    test('Panel testing example', async () => {
+      // Example: Testing panel functionality
+      render(
+        <CodeEditor
+          defaultValue="SELECT * FROM users;"
+          panel={
+            <Panel
+              title="SQL Query"
+              showCopyButton={true}
+              showFormatButton={true}
+              data-lgid="lg-panel-example"
```

Comment:
Oh, I see. You're passing the same data-lgid to `CodeEditor` and `Panel`, so `Panel` has the correct prefix. You can avoid this if you use the context approach I mentioned above.